### PR TITLE
Add setting for captcha disable and MISC

### PR DESCRIPTION
**Pull request by abbychau**
update env example for to 127.0.0.1
(localhost is resolver-dependent
could cause SQLSTATE[HY000] [2002] error on artisan installation)

add git ignore for IJ and R
Add settings for captcha disable 

More localizations

some layout fixes

### CODE REVIEW

File: app/Composers/AppComposer.php [hunk 1]
```diff
@@ -32,5 +32,7 @@ public function compose(View $view)
         $view->withSiteLogo(Config::get('setting.site_logo'));
         $view->withSiteTitle(Config::get('setting.site_name').' | Hifone');
         $view->withSiteLocale(Config::get('setting.site_locale'));
+        $view->withSiteCaptchaRegDisabled(Config::get('setting.site_captcha_reg_disabled'));
```
Comment (lianguan) [inline app/Composers/AppComposer.php]: 这两个变量并非整站都需要，不建议在此赋值。建议在相应的controller里做。

File: app/Http/Controllers/Auth/AuthController.php [hunk 2]
```diff
@@ -83,7 +84,7 @@ public function postLogin()
         $loginData = Input::only(['login', 'password', 'verifycode']);
 
         $verifycode = array_pull($loginData, 'verifycode');
-        if ($verifycode != Session::get('phrase')) {
+        if ($verifycode != Session::get('phrase') && Config::get('setting.site_captcha_login_disabled')) {
```
Comment (lianguan) [inline app/Http/Controllers/Auth/AuthController.php]: 建议修改为：

``` php
if(!Config::get('setting.site_captcha_login_disabled')) {
 $verifycode = array_pull($loginData, 'verifycode');
....
}
```

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,6 @@
 .env
 node_modules
 phpunit.xml
-vendor
\ No newline at end of file
+vendor
+.idea/
+.Rhistory
```
Comment (lianguan) [inline .gitignore]: .idea可以加上，建议删除.Rhistory，此处只考虑大多数人的开发环境。

**Comment by abbychau** [comment]
Can you suggest the controllers which I should consider putting $view->withSiteCaptcha.... ?

**Comment by lianguan** [comment]
@abbychau 
AuthController is the best. :laughing:

**Comment by lianguan** [comment]
nice. thanks.

**Comment by abbychau** [comment]
p.s. AuthController is not yet changed.

**Comment by lianguan** [comment]
@abbychau  I did it. :laughing:

**Comment by abbychau** [comment]
Whisp great :+1: .. I was steathily visiting GitHub during office hour. I can only check out at home. (Sweating:p

**Pull request merged**